tests/blockchain/regtest/SendToPaymentCode.cpp

```diff
@@ -0,0 +1,105 @@
+#include "ottest/fixtures/blockchain/RegtestSimple.hpp"
+
+namespace ottest
+{
+
+TEST_F(Regtest_fixture_simple, send_to_payment_code)
+{
+    EXPECT_TRUE(Start());
+    EXPECT_TRUE(Connect());
+
+    const std::string name_alice = "Alice";
+    const std::string name_bob = "Bob";
+
+    Height begin = 0;
+    const auto blocks_number = 2;
+    auto coin_to_send = 100000;
+
+    auto [user_alice, success_alice] = CreateClient(
+        opentxs::Options{},
+        3,
+        core_wallet_.name_,
+        core_wallet_.words_,
+        address_);
+    EXPECT_TRUE(success_alice);
+
+    auto [user_bob, success_bob] = CreateClient(
+        opentxs::Options{},
+        4,
+        name_bob,
+        GetPaymentCodeVector3().bob_.words_,
+        address_);
+    EXPECT_TRUE(success_bob);
+
+    std::vector<std::reference_wrapper<const User>> users{user_alice, user_bob};
+    MineBlocksForUsers(users, target_height, blocks_number);
+    begin = target_height;
+    target_height += static_cast<int>(MaturationInterval()) + 1;
+
+    // mine MaturationInterval number block with
+    MineBlocks(begin, static_cast<int>(MaturationInterval()) + 1);
+    begin += MaturationInterval() + 1;
+
+    EXPECT_EQ(
+        GetBalance(user_alice),
+        amount_in_transaction_ * blocks_number * transaction_in_block_);
+
+    EXPECT_EQ(
+        GetBalance(user_bob),
+        amount_in_transaction_ * blocks_number * transaction_in_block_);
+
+    User* sender = &users_.at(name_alice);
+    User* receiver = &users_.at(name_bob);
+    sender->expected_balance_ =
+        amount_in_transaction_ * blocks_number * transaction_in_block_;
+    receiver->expected_balance_ =
+        amount_in_transaction_ * blocks_number * transaction_in_block_;
+
+    const size_t numbers_of_test = 2;
+
+    for (size_t number_of_test = 0; number_of_test < numbers_of_test;
+         number_of_test++) {
+
+        const auto& network =
+            sender->api_->Network().Blockchain().GetChain(test_chain_);
+
+        auto future = network.SendToPaymentCode(
+            sender->nym_id_,
+            receiver->PaymentCode(),
+            coin_to_send,
+            "memo for outgoing transaction");
+
+        MineTransaction(*sender, future.get().second, target_height);
+
+        auto loaded_transactions = CollectTransactionsForFeeCalculations(
+            *sender, send_transactions_, transactions_ptxid_);
+        auto fee = CalculateFee(send_transactions_, loaded_transactions);
+        send_transactions_.clear();
+
+        sender->expected_balance_ -= Amount{coin_to_send} + fee;
+        receiver->expected_balance_ += coin_to_send;
+
+        WaitForSynchro(*sender, target_height, sender->expected_balance_);
+        WaitForSynchro(*receiver, target_height, receiver->expected_balance_);
+
+        EXPECT_EQ(GetBalance(*sender), sender->expected_balance_);
+        EXPECT_EQ(GetBalance(*receiver), receiver->expected_balance_);
+
+        std::swap(sender, receiver);
```

Comment:
i assume You want to do same thing twice but with swaped sender and receiver ? Maybe method would be better way to go ? instead of loop ?

Reply:
we want to do same operations as above but for different user. By using swap you make sure that you will run out of coins later (we just losing fees and not coins)

Reply:
yes i know but whole block is runned twice just only 1 difference sender vs receiver, method imho wouyld be advised

Reply:
but then how it would looks like?
for
{
if(i % 0)
do_stuff(bob, alice);
else
do_stuff(alice, bob)

}
?:D

Reply:
ok probably im not understood problem, since i believe it would be just : 
{
do_stuff(bob, alice);
do_stuff(alice, bob);
}